src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOps.java

```diff
@@ -2791,6 +2791,53 @@ public Value processor() {
         public List<Value> fragments() {
             return operands().subList(1, operands().size());
         }
+
+        public List<Body> expressions() {
+            return expressions;
+        }
+
+        @Override
+        public Block.Builder lower(Block.Builder b, OpTransformer opT) {
+
+            Block.Builder[] builders = new Block.Builder[1];
+            builders[0] = b;
+            List<Value> expressions = new ArrayList<>();
+            for (Body expression : expressions()) {
+                builders[0].transformBody(expression, List.of(), opT.andThen((block, op) -> {
+                    if (op instanceof YieldOp yop) {
+                        expressions.add(block.context().getValue(yop.yieldValue()));
+                        builders[0] = block;
```

Comment:
Making a note for future change. This code suggests that `transformBody` could return the last block builder, so you don't have to keep track of it yourself. Another solution is to create blocks for the start of each expression and replace the yield with a branch to the next expression.

Reply:
I implemented the other solution.